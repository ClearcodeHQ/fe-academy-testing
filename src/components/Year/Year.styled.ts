import styled from 'styled-components';

export const Container = styled.div`
  margin: 0 auto;
  width: 200px;
  padding: 10px;
  border: 1px solid #ffcd32;
  margin-bottom: 8px;
`;
