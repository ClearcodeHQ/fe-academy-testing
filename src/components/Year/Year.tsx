import React from 'react';
import { Container } from './Year.styled';

export const Year = () => {
  return (
    <Container>
      <span>Year: {new Date().getFullYear()}</span>
    </Container>
  );
};
