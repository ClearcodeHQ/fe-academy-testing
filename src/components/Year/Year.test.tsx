import { render, screen } from '@testing-library/react';
import React from 'react';
import { Year } from './Year';

describe('Year', () => {
    beforeAll(() => {
      jest.useFakeTimers();
      jest.setSystemTime(new Date('2022-01-01'));
    });

    afterAll(() => {
      jest.useRealTimers();
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(<Year />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should render 2022 year',  () => {
      render(<Year />);

      const year = screen.getByText(/Year:/);

      expect(year).toHaveTextContent('Year: 2022');
    });
  });
