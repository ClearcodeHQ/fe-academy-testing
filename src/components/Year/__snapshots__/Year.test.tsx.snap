// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Year should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ hHQYvn"
  >
    <span>
      Year: 2022
    </span>
  </div>
</DocumentFragment>
`;
